test/built-ins/Proxy/defineProperty/targetdesc-not-configurable-writable-desc-not-writable.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019 Aleksey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 9.5.6
+description: >
+    Throw a TypeError exception if trap result is true, targetDesc is not configurable
+    and writable, while Desc is not writable.
+info: |
+    [[DefineOwnProperty]] (P, Desc)
+
+    ...
+    16. Else targetDesc is not undefined,
+        ...
+        c. If IsDataDescriptor(targetDesc) is true, targetDesc.[[Configurable]] is
+        false, and targetDesc.[[Writable]] is true, then
+            i. If Desc has a [[Writable]] field and Desc.[[Writable]] is
+            false, throw a TypeError exception.
+    ...
+features: [Proxy]
+---*/
+
+var target = {};
+var p = new Proxy(target, {
+  defineProperty: function(t, prop, desc) {
+    return true;
+  },
+});
+
+Object.defineProperty(target, "prop", {
+  configurable: false,
+  writable: true,
+});
+
+assert.throws(TypeError, function() {
+  Object.defineProperty(p, "prop", {
+    writable: false,
+  });
+});
```

Comment:
It would be nice to add another assertion where the property is defined into target inside the proxy trap, i.e.:

```js
var trappedExecution = 0;
var p = new Proxy({}, {
  defineProperty: function(t, prop, desc) {
    Object.defineProperty(target, 'prop', {
      configurable: false,
      writable: true,
    });
    trappedExecution += 1;
    return true;
  },
});

assert.throws(TypeError, function() {
  Reflect.defineProperty(p, 'prop', {
    writable: false,
  });
});
assert.sameValue(trappedExecution, 1); // We make sure the operation was trapped from the proxy and the property set to target.
```

Reply:
79d97ae